plugins/turing/runner/experiment_runner.go

```diff
@@ -44,67 +49,101 @@ func (er *experimentRunner) GetTreatmentForRequest(
+		return &runner.Treatment{
+			Config: nil,
+		}, nil
 	}
-	if treatmentResponse.JSON400 != nil {
-		return nil, fmt.Errorf(treatmentErrTpl, treatmentResponse.JSON400.Message)
+
+	randomizationKeyValue, err := er.appContext.SchemaService.GetRandomizationKeyValue(projectId, requestParams)
+	if err != nil {
+		return nil, err
 	}
-	if treatmentResponse.JSON500 != nil {
-		return nil, fmt.Errorf(treatmentErrTpl, treatmentResponse.JSON500.Message)
+
+	selectedTreatment, switchbackWindowId, err = er.appContext.TreatmentService.GetTreatment(filteredExperiment, randomizationKeyValue)
+	if err != nil {
+		return nil, err
 	}
-	if treatmentResponse.JSON200 == nil {
-		return nil, fmt.Errorf(treatmentErrTpl, "empty response body")
+
+	treatmentRepr := models.ExperimentTreatmentToOpenAPITreatment(selectedTreatment)
+
+	// Marshal and return response
+	treatment := schema.SelectedTreatment{
+		ExperimentId:   filteredExperiment.Id,
+		ExperimentName: filteredExperiment.Name,
+		Treatment:      treatmentRepr,
+		Metadata: schema.SelectedTreatmentMetadata{
+			ExperimentVersion:  filteredExperiment.Version,
+			ExperimentType:     models.ProtobufExperimentTypeToOpenAPI(filteredExperiment.Type),
+			SwitchbackWindowId: switchbackWindowId,
+		},
 	}
 
 	// Marshal Response Body
-	rawConfig, err := json.Marshal(treatmentResponse.JSON200.Data)
+	rawConfig, err := json.Marshal(treatment)
 	if err != nil {
 		return nil, fmt.Errorf("Error marshalling the treatment config: %s", err.Error())
 	}
 
-	// Return treatment info
-	var expName, treatmentName string
-	if treatmentResponse.JSON200.Data != nil {
-		expName = treatmentResponse.JSON200.Data.ExperimentName
-		treatmentName = treatmentResponse.JSON200.Data.Treatment.Name
-	}
 	return &runner.Treatment{
-		ExperimentName: expName,
-		Name:           treatmentName,
+		ExperimentName: filteredExperiment.Name,
+		Name:           selectedTreatment.Name,
 		Config:         rawConfig,
 	}, nil
 }
 
+func (er *experimentRunner) startBackgroundServices(
+	errChannel chan error,
+) {
+	backgroundSvcCtx := context.Background()
+	if er.appContext.ExperimentSubscriber != nil {
+		go func() {
+			err := er.appContext.ExperimentSubscriber.SubscribeToManagementService(backgroundSvcCtx)
+			if err != nil {
+				errChannel <- err
+			}
+		}()
+	}
+	return
+}
+
 func (er *experimentRunner) getRequestParams(
 	logger log.Logger,
 	reqHeader http.Header,
 	body []byte,
-) map[string]string {
+) map[string]interface{} {
 	// Get the request parameters for the current request
-	requestParams := map[string]string{}
+	requestParams := map[string]interface{}{}
 	for _, param := range er.parameters {
 		if param.FieldSource == "none" || param.Field == "" {
 			// Parameter not configured
 			continue
 		}
-		val, err := request.GetValueFromRequest(reqHeader, body, request.FieldSource(param.FieldSource), param.Field)
+		val, err := request.GetValueFromHTTPRequest(reqHeader, body, request.FieldSource(param.FieldSource), param.Field)
```

Comment:
Updated this when I bumped up the dependency on `github.com/caraml-dev/turing/engines/experiment`.